fbt_lib/src/lib.rs

```diff
@@ -1,13 +1,22 @@
+use crate::types::{Failure, SingleTestResult, TestResult};
 use std::path::PathBuf;
+use std::time::{Duration, Instant};
 
 pub mod types;
 
-pub fn test_all() -> anyhow::Result<bool> {
+pub fn test_all() -> anyhow::Result<TestResult> {
```

Comment:
Create custom error type using `thiserror` library. Refer [this blog post](https://nick.groenen.me/posts/rust-error-handling/).